metals/src/main/scala/scala/meta/internal/metals/codelenses/RunTestCodeLens.scala

```diff
@@ -315,8 +336,8 @@ final class RunTestCodeLens(
     val (data, shellCommandAdded) =
       if (!isJVM) (main.toJson, false)
       else
-        buildTargetClasses.jvmRunEnvironment
-          .get(target)
+        buildTargetClasses
+          .jvmRunEnvironmentSync(target)
```

Comment:
Just a question, can we ask for jvmRunEnvironment here instead? This would remove the need for `jvmRunEnvironmentSync`, since `jvmRunEnvironment` can return cached results.

Or maybe waiting for the response here would be too long?

Reply:
I would rather avoid using Await here and using Future propagates it all the way and it becomes quite complex. There are multiple sequences of futures etc.